### PR TITLE
UI complete

### LINKED ISSUE #5: Amélioration tableau joueur
Ajout de colorPicker en face de chaque ligne du tableau de création des joueurs

### LINKED ISSUE #6: Modifier PlateauParameter
Ajouter la liste des joueurs dans le PlateauParameter

### PR DESCRIPTION
**Pull request by DoBrcd**

### CODE REVIEW

File: src/model/Columns.java [hunk 1]
```diff
@@ -12,41 +12,49 @@
...
     }
 
-    public List<Checker> getCheckers(){
-        return Checkers;
+    public List<Checker> getCheckers() {
+        return checkers;
     }
 
-    public int getHauteur(){
+    public int getHauteur() {
```
Comment (Enoriz) [inline src/model/Columns.java]: in english please

File: src/model/Columns.java [hunk 2]
```diff
@@ -12,41 +12,49 @@
...
         return hauteur;
     }
 
-    //Autres
-    public void Empile(Checker Checker) {
-        Checkers.add(Checker);
+    // Autres
+    public void Empile(Checker checker) {
```
Comment (Enoriz) [inline src/model/Columns.java]: in english please

File: src/model/Game.java [hunk 3]
```diff
@@ -57,14 +57,23 @@ public int getNbPlayer(){
         return nbPlayer;
     }
 
-    public void JouerPion(int colonneIndex){
-        this.gameBoard.JouerPion(playerList.get(actualPlayer), colonneIndex);
-        actualPlayer = actualPlayer++ % playerList.size();
+    private boolean jouerPion(Joueur j, int colonneIndex) {
```
Comment (Enoriz) [inline src/model/Game.java]: in english please

File: src/model/Game.java [hunk 4]
```diff
@@ -57,14 +57,23 @@ public int getNbPlayer(){
...
+            return true;
+        }
+        return false;
     }
 
-    //Modificateur
-    //D�gueu mais elle ne sera pas trop utilis�e OU a changer
-    public void setActualPlayer(int newActualPlayer){
-        actualPlayer = newActualPlayer;
+    public boolean JouerPion(int colonneIndex) {
```
Comment (Enoriz) [inline src/model/Game.java]: in english please

File: src/model/GameBoard.java [hunk 5]
```diff
@@ -45,28 +45,126 @@ public GameBoard(int nbColonne, int hauteurColonne) {
...
-    //Autres
-    public void JouerPion(Joueur joueur, int colonneIndex) {
+
+	public List<Columns> getColumns() {
+		return colonnes;
+	}
+
+    
+    // Autres
+    public boolean JouerPion(Joueur joueur, int colonneIndex) {
```
Comment (Enoriz) [inline src/model/GameBoard.java]: in english please, oh et la classe joueur aussi x)

File: src/model/Joueur.java [hunk 6]
```diff
@@ -16,15 +16,19 @@
     private Color color;
 
     //Constructeur
-    public Joueur (String name ,String color){
+    public Joueur (String name ,Color color){
```
Comment (Enoriz) [inline src/model/Joueur.java]: in english please

File: src/application/GameBoard.java [hunk 7]
```diff
@@ -1,52 +1,142 @@
...
+        }
+    }
+
+    private void createBoard(BoardParameters params) {
+        createBoard(params.getNbRow(), params.getNbCol());
+    }
+
+    // Création du plateau de jeu avec possibilité de modifier le nombee de
+    // ligne/colonne
+    private void createBoard(int row, int columns) {
```
Comment (zartech22) [inline src/application/GameBoard.java]: row -> rows

File: src/application/GameBoard.java [hunk 8]
```diff
@@ -1,52 +1,142 @@
...
+                cercle.setFill(checker.getColor());
+            } else {
                 cercle.setFill(Color.WHITE);
-                root.getChildren().add(cercle);
             }
+            vbox.getChildren().add(cercle);
         }
+    }
 
+    private void clicOnColums(VBox vBox) {
```
Comment (zartech22) [inline src/application/GameBoard.java]: clickOnColumn

File: src/application/MainFrame.java [hunk 9]
```diff
@@ -58,4 +71,44 @@ public void afficheTableauScore(){
             e.printStackTrace();
         }
     }
+
+    public void startGame(int row, int columns){
+        try {
+            FXMLLoader loader = new FXMLLoader();
+            loader.setLocation(Main.class.getResource("../FXML/GameBoard.fxml"));
```
Comment (zartech22) [inline src/application/MainFrame.java]: use this.class()

File: src/application/MainFrame.java [hunk 10]
```diff
@@ -58,4 +71,44 @@ public void afficheTableauScore(){
...
+
+            SplitPane page = loader.load();
+            Scene scene = new Scene(page);
+
+            this.setScene(scene);
+            this.sizeToScene();
+
+            GameBoard ctrl = loader.getController();
+            BoardParameters params = new BoardParameters(row, columns);
```
Comment (zartech22) [inline src/application/MainFrame.java]: row -> rows

File: src/application/NewGameDialog.java [hunk 11]
```diff
@@ -1,65 +1,89 @@
...
 
-        nomCol.setCellFactory(TextFieldTableCell.<Joueur>forTableColumn());
-        couleurCol.setCellFactory(TextFieldTableCell.<Joueur>forTableColumn());
+        joueurTableView.setItems(parameters.getPlayers());
 
-        joueurTableView.setItems(list);
+        parameters.getNbColProperty().bind(nbColonnes.valueProperty());
+        parameters.getNbRowProperty().bind(nbLignes.valueProperty());
+
+        btnJouer.setOnMouseClicked(e -> MainFrame.getMainFrame().startGame(parameters));
```
Comment (zartech22) [inline src/application/NewGameDialog.java]: In english

File: src/application/WelcomeStage.java [hunk 12]
```diff
@@ -13,12 +13,10 @@
 public class WelcomeStage extends Scene {
     private Button nouvellePartie, afficherScores, quitter;
```
Comment (zartech22) [inline src/application/WelcomeStage.java]: In english

File: src/model/BoardParameters.java [hunk 13]
```diff
@@ -1,32 +1,57 @@
...
+     *
+     * @param row
+     * @param columns
      */
-    public BoardParameters(int nbColonne, int hauteurColonne) {
-        this.nbColonne = nbColonne;
-        this.hauteurColonne = hauteurColonne;
+    public BoardParameters(int row, int columns) {
+        this.nbColumns = new SimpleIntegerProperty(columns);
+        this.nbRow = new SimpleIntegerProperty(row);
```
Comment (zartech22) [inline src/model/BoardParameters.java]: Row -> Rows

File: src/model/Game.java [hunk 14]
```diff
@@ -12,32 +9,32 @@
  */
 public class Game {
     private GameBoard gameBoard;
-    private List<Joueur> playerList;
+    private List<Player> playerList;
     private int actualPlayer, nbPlayer;
```
Comment (zartech22) [inline src/model/Game.java]: actualPlayer -> currentPlayer

**Review by zartech22 — CHANGES_REQUESTED** [review]
Petites modifications

**Review by zartech22 — APPROVED** [review]

**Pull request merged**